### DISCUSSION TITLE
is this way of implementing props ok ?

### DISCUSSION THREAD

**Discussion by robertpanvip** [solidjs/solid#2335] [Ideas]
### Why not design it that way 
```language
function MyComponent(props) {
  const {name} = props
  return <div>Hello {props.name()}---{name()}</div>;
}

<MyComponent name="Solid" />;
```

So it's designed with the signal  Also easy to deconstruct

**Comment by atk** [comment]
The main reason is that this can easily get confusing if the prop contains a function and one forgets to call twice.

If your react habits die hard, you could try https://github.com/orenelbaum/babel-plugin-solid-undestructure.
